### PR TITLE
improves airlock sounds via porting part of RP's sound overhaul + toggle

### PR DESCRIPTION
**Pull request by buffyuwu**
## About The Pull Request

ports part of https://github.com/Citadel-Station-13/Citadel-Station-13-RP/pull/2648#issue-560565262

each airlock will have a unique sound based on it's type. security airlocks sound one way, command airlocks sound another. the sounds are much less grindy and high pitched ear rape, and bring more of an immersive feel overall.

by default the airlock sounds are the 'classic' ones, but you can toggle this update on with the 'toggle airlock sound type' verb in the 'preferences' tab.

demonstration:
https://streamable.com/gfdv7l

toggle:
https://user-images.githubusercontent.com/42501819/137806300-5842dce7-1f8c-41aa-865e-37d3e43a86ba.mp4

## Changelog
:cl:
soundadd: ported unique airlock sounds from RP
tweak: airlock sounds can now be toggled to a new version in the preferences tab
/:cl:

### CODE REVIEW

**Comment by RummyRin** [comment]
Just as long as old airlock is a toggle option this seems fine.

**Comment by Hatterhat** [comment]
make it a toggle i want my Soul

**Comment by TheEaterOfBeans** [comment]
I will never get used to a different SS13 airlock sound.

**Comment by WolfTreee** [comment]
Please, to space gods, MAKE A toggle or revert this. These are far, far worse than before.

**Comment by Fellout2** [comment]
I agree with the general feel on the sounds being a toggle, that would appease those who like the old sounds and the new ones

**Comment by Tupinambis** [comment]
Yeah these are otherwise too disruptive imo. The sounds are too detailed to the point of being distracting.

**Comment by buffyuwu** [comment]
yeah i'll look into making this a toggle i forgot this existed until it got testmerged

**Comment by buffyuwu** [comment]
> Yeah these are otherwise too disruptive imo. The sounds are too detailed to the point of being distracting.

> I agree with the general feel on the sounds being a toggle, that would appease those who like the old sounds and the new ones

> Please, to space gods, MAKE A toggle or revert this. These are far, far worse than before.

> I will never get used to a different SS13 airlock sound.

> Just as long as old airlock is a toggle option this seems fine.


https://user-images.githubusercontent.com/42501819/137697296-dae27b96-9914-4129-9662-8c613b8c5be9.mp4

**Comment by Fellout2** [comment]
So why is it removed from even being a toggle? I preferred these sounds over the old ones so much, made the station feel more alive and fun to walk around in

**Comment by buffyuwu** [comment]
this has a toggle and is sitting in my github pull requests triggering my ocd can someone review this please :)

File: code/modules/client/preferences_toggles.dm [hunk 1]
```diff
@@ -199,6 +199,20 @@ TOGGLE_CHECKBOX(/datum/verbs/menu/Settings/Sound, toggle_ship_ambience)()
...
+	set name = "Toggle Airlock Sound Type"
+	set category = "Preferences"
+	set desc = "Old or new."
+	usr.client.prefs.toggles ^= SOUND_AIRLOCKS
+	usr.client.prefs.save_preferences()
+	if(usr.client.prefs.toggles & SOUND_AIRLOCKS)
+		to_chat(usr, "You will now hear alternate airlock sounds.")
+	else
+		to_chat(usr, "You will no longer hear alternate airlock sounds.")
+	SSblackbox.record_feedback("nested tally", "preferences_verb", 1, list("Toggle Airlock Sounds", "[usr.client.prefs.toggles & SOUND_AIRLOCKS ? "Enabled" : "Disabled"]")) // im sorry this didnt get testmerged with the pref at the beginning
```
Comment (DeltaFire15) [inline code/modules/client/preferences_toggles.dm:212]: Ambiguous blackbox record with no true information gain - might be one person toggling repeatedly, or many toggling once. Unneccessary.

Comment (buffyuwu) [reply code/modules/client/preferences_toggles.dm:212]: The entire file has the same style of recording feedback for hitting these audio toggles. If you want to remove these feedbacks from being recorded, open another PR.

**Comment by deathride58** [comment]
I know this PR is months old, but uh. The sounds here are legitimately pretty grating. Most of them sustain for an ungodly amount of time, and have a harsh cutoff. Further, some of them, such as medical airlocks, lack a tactile punch.

There's also minor nitpicks; the way the pref's handle breaks compatibility with multiz audio, for instance, but since airlocks in multiz maps are rarely exposed to openspaces, this isn't too big a deal.

The folks on RP; do they seriously live like this?

**Pull request closed without merging**